x-pack/filebeat/input/s3/input.go

```diff
@@ -0,0 +1,400 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package s3
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"strings"
+	"sync"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/defaults"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/harvester"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	// Filebeat input name
+	inputName = "s3"
+
+	// The maximum number of messages to return. Amazon SQS never returns more messages
+	// than this value (however, fewer messages might be returned).
+	maxNumberOfMessage int64 = 10
+
+	// The duration (in seconds) for which the call waits for a message to arrive
+	// in the queue before returning. If a message is available, the call returns
+	// sooner than WaitTimeSeconds. If no messages are available and the wait time
+	// expires, the call returns successfully with an empty list of messages.
+	waitTimeSecond int64 = 10
+
+	// The duration (in seconds) that the received messages are hidden from subsequent
+	// retrieve requests after being retrieved by a ReceiveMessage request.
+	// This value needs to be a lot bigger than filebeat collection frequency so
+	// if it took too long to read the s3 log, this sqs message will not be reprocessed.
+	// The default visibility timeout for a message is 30 seconds. The minimum
+	// is 0 seconds. The maximum is 12 hours.
+	visibilityTimeout int64 = 300
```

Comment:
Question, should these parameters be a part of the s3 input parameters instead of hardcoded here?

Reply:
some of them may make sense? let's try to figure out once this is working

Reply:
Based on logstash input for s3-sqs, these are input parameters 😬

Reply:
+1 to add the ones that you feel make sense

Reply:
Done 😬